libraries/javacheck/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ find_package(Java 1.7 REQUIRED COMPONENTS Development)
 
 include(UseJava)
 set(CMAKE_JAVA_JAR_ENTRY_POINT JavaCheck)
-set(CMAKE_JAVA_COMPILE_FLAGS -target 8 -source 8 -Xlint:deprecation -Xlint:unchecked)
+set(CMAKE_JAVA_COMPILE_FLAGS -target 7 -source 7 -Xlint:deprecation -Xlint:unchecked)
```

Comment:
You're gonna have to add jdk7 to the GHA too.

Reply:
no, this way you're just compiling against java 7, you don't need to use java 7, we're already doing this for newlaunch

Reply:
As already mentioned, javac [supports](https://docs.oracle.com/en/java/javase/17/docs/specs/man/javac.html#compiling-for-earlier-releases-of-the-platform) compilation for Java releases earlier then the release of JDK javac belongs to.